mlir/utils/jenkins/Jenkinsfile

```diff
@@ -187,6 +190,9 @@ pipeline {
                             alwaysPull true
                         }
                     }
+                    environment {
+                        HOME="${WORKSPACE}"
```

Comment:
Why?

Reply:
Our dockerfile set HOME to /root.

Reply:
I think the better fix here is to not do that in the Dockerfile, or to unset HOME altogether. Do we _need_ a value of $HOME in this build?

Reply:
Agreed, on a second look I think `ENV HOME /root/` was a mistake I made (in combination with `-u 0`). I'm not sure why I explicitly set the root back then but it does seem to be a more robust fix to get rid of it al-together in the dockerfile.

Reply:
MIOpen needs to access $HOME when installing dependences (cget and sqlite). We can unset HOME in the Dockerfile, but we still need to set HOME here, since the default HOME for a jenkins user would be /home/jenkins (owned by root), while we want /home/jenkins/$JOBNAME, which is $WORKSPACE

Reply:
I take it there isn't a way to do this in the Docker launch?

Reply:
Not sure if '-e' would work. Another way is to add a parameter using ARG to the dockerfile, but then we need to rebuild docker every time when we start a docker container.

Reply:
@yiqian1 Can you try adding `-e HOME=${WORKSPACE}` (using double quotes for the Docker args) instead of these environment lines?

Reply:
Just to check, did you try `-e HOME=${WORKSPACE}`?

Reply:
Yes. You can see what I've tried [here](https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/commits/testHome/mlir/utils/jenkins/Jenkinsfile)

Reply:
http://rocmhead.amd.com:8080/job/testHome/2/console
groovy.lang.MissingPropertyException: No such property: WORKSPACE for class: groovy.lang.Binding

Reply:
Maybe you need to first define HOME as an environment variable before setting it. So maybe test it by locally adding a line of `ENV HOME /home`, and then reset it in invocation time, for example.

Reply:
Nah, that's a Jenkins-side error.

One way to clean things up would be to make dockerArgs() take a `String homeDir` .parameter and pass in `"${WORKSPACE}"` there so that we're not having Jenkins set environment variables that could be reasonably delegated to the Docker container startup.

Reply:
No luck with adding a dockerArgs() parameter. Got the same error:
groovy.lang.MissingPropertyException: No such property: WORKSPACE for class: groovy.lang.Binding

Reply:
What I meant was that instead of substituting in WORKSPACE, to add a parameter to the function, substitute in that parameter, and then generally set that parameter to WORKSPACE (or even something interesting like /tmp)

Reply:
That was exactly what I did in [this commit](https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/commit/2225ab54e3a65096b40c0b42cdb148294f5799aa#diff-16af82a543140fc9970902655eb9815da207d857b849d253ab13c4c0f344aa6a)

And the jenkins job failed http://rocmhead.amd.com:8080/job/testHome/9/console